internal/auth/auth.go

```diff
@@ -1,44 +1,70 @@
 package auth
 
 import (
-	"fmt"
-	"github.com/TheDevExperiment/server/internal/db/repositories"
-	"github.com/TheDevExperiment/server/router/models"
-	"github.com/gin-gonic/gin"
-	"go.mongodb.org/mongo-driver/bson"
+	"log"
 	"net/http"
+
+	"github.com/TheDevExperiment/server/internal/db/users"
+	userModels "github.com/TheDevExperiment/server/internal/db/users/models"
+	"github.com/TheDevExperiment/server/middlewares"
+	"github.com/TheDevExperiment/server/router/models/auth"
+	"github.com/gin-gonic/gin"
 )
 
 func GuestValidateV1(c *gin.Context) {
-	// first bind the req to our model
-	var req models.AuthRequest
-	var res models.AuthResponse
-	userRepository := c.MustGet("userRepository").(*repositories.UserRepository)
+	var res auth.AuthResponse
```

Comment:
```suggestion
	var res auth.Response
```
golang naming convention best practice. Since Response is already inside auth package, it makes it obvious that its AuthResponse. So don't include package name in your variables